src/buf_writer.c

```diff
@@ -112,10 +118,19 @@ buf_writer_cb(void *buf_writer_arg, const char *s) {
 	buf_writer_assert(buf_writer);
 	assert(buf_writer->buf != NULL);
 	assert(buf_writer->buf_end <= buf_writer->buf_size);
+	if (buf_writer->overflowed) {
+		return;
+	}
 	size_t i, slen, n, s_remain, buf_remain;
 	for (i = 0, slen = strlen(s); i < slen; i += n) {
 		if (buf_writer->buf_end == buf_writer->buf_size) {
-			buf_writer_flush(buf_writer);
+			if (buf_writer->allow_auto_flush) {
+				buf_writer_flush(buf_writer);
+			} else {
+				buf_writer->buf_end = buf_writer->committed;
+				buf_writer->overflowed = true;
+				return;
```

Comment:
nit: move the overflowed branch to the s_remain < buf_remain below, to avoid the memcpy of the last iteration.

Reply:
We can only copy it but not move it. We still need to test at the beginning. Flushing is by design lazy in buffered writer, so if in a previous `write_cb()` call the buffer was exactly full, then the initial test is necessary. I think having a single test at the beginning outweighs the saving of one `memcpy`. Thoughts?

Reply:
Yeah keep the branch at beginning, but add another one right before memcpy, so that we can set overflow and return immediately.